cmd/incus/top.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	cli "github.com/lxc/incus/v6/internal/cmd"
+	"github.com/lxc/incus/v6/internal/i18n"
+	"github.com/lxc/incus/v6/internal/server/metrics"
+)
+
+type cmdTop struct {
+	global *cmdGlobal
+}
+
+// Command is a method of the cmdTop structure that returns a new cobra Command for managing command aliases.
+func (c *cmdTop) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = usage("top")
+	cmd.Short = i18n.G("Manage command aliases")
+	cmd.Long = cli.FormatSection(i18n.G("Description"), i18n.G(
+		`Manage command aliases`))
+
+	// Workaround for subcommand usage errors. See: https://github.com/spf13/cobra/issues/706
+	cmd.Args = cobra.NoArgs
+	cmd.RunE = c.Run
+	return cmd
+}
+
+// Run is a method of the cmdTop structure. It implements the logic to call `incus top`.
+// This function implements the `top` command. It queries the metrics API at (/1.0/metrics) and renders a list of
+// instances with their CPU, memory and disk usage columns
+func (c *cmdTop) Run(cmd *cobra.Command, args []string) error {
+	conf := c.global.conf
+
+	remote, path, err := conf.ParseRemote("/1.0/metrics")
```

Comment:
So I'm not too sure what's going on here.
`conf.ParseRemote` should be passed `args[0]` as is usual as that just points it to the correct server (e.g. `incus top my-cluster:`)

Reply:
The `path` part is actually irrelevant in this case, you only really need `remote`.

Reply:
If no argument is passed, should we assume that the remote is local then?

Reply:
Yeah, you can look at `monitor.go` for an example command with similar argument handling.